roles/users/tasks/main.yml

```diff
@@ -36,7 +36,7 @@
     password: '{% for user in vault_users | default([]) if user.name == item.name and user.password is defined %}{{ user.password | password_hash("sha512", user.salt | default("") | truncate(16, true, "") | regex_replace("[^\.\/a-zA-Z0-9]", "x")) }}{% else %}{{ None }}{% endfor %}'
     state: present
     shell: /bin/bash
-    update_password: always
+    update_password: "{{ item.value.update_password | default('always') }}"
   with_items: "{{ users }}"
```

Comment:
Given the `with_items` loop (not `with_dict`), I believe it should just be
`item.update_password`
(vs. `item.value.update_password`)